scpproxy/src/main.rs

```diff
@@ -182,18 +183,25 @@ fn run(scu_stream: &mut TcpStream, destination_addr: &str) -> Result<()> {
                     Ok(())
                 });
             }
+            let mut buffer: Vec<u8> = Vec::with_capacity(MINIMUM_PDU_SIZE as usize);
 
             loop {
                 let message = message_rx.recv().context(ReceiveMessage)?;
                 match message {
                     ThreadMessage::SendPdu { to, pdu } => match to {
                         ProviderType::Scu => {
-                            println!("scu <---- scp: {:?}", &pdu);
-                            write_pdu(scu_stream, &pdu).unwrap();
+                            // println!("scu <---- scp: {:?}", &pdu);
```

Comment:
Printing the whole thing is indeed far from ideal. But we could make a compromise with a shortened PDU display implementation. Stating the kind of PDU alone would be rather helpful. This could either be done directly on the `Pdu` type or with a newtype designed for creating a short human readable representation.

Reply:
I did add a `pdu.short_description()`  and wrapped `if verbose { ... }` around the printings.